### PR TITLE
ME-2744: connector install flow should write token to config file

### PR DESCRIPTION
**Pull request by waltzofpearls**
# Description

tldr: this change helps restarting docker container based connector with or without `--invite` flag.

invite code based connector install flow, particularly used by the one-liner docker run command that starts the connector with an invite code, should also write connector token to the border0 config file, which is located at `$HOME/.border0/config.yml`.

previously, in the connector install flow, connector token was only written to files with invite code as file names, for example, `$HOME/.border0/invite_<invite_code>`. with that, to restart the same container, user has to run the same command with the exactly the same invite code, but if the user is not aware of that, there is no way to retrieve the invite code, and then get stuck and confused

with this new change, user can start the container again with or without the invite code.

the invite code is convenient for starting up multiple docker containers that run border0 connector, that said, most users likely only need to run one connector with docker on a computer.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: internal/connector_v2/invite/invite.go [hunk 1]
```diff
@@ -85,14 +87,22 @@ func readConnectorTokenFromFile(inviteCode string) (string, error) {
...
+	border0ConfigFile := filepath.Join(dotBorder0Dir, "config.yaml")
+	yamlBytes, err := yaml.Marshal(config.Configuration{Token: connectorToken})
+	if err != nil {
+		return fmt.Errorf("failed to marshal configuration to yaml: %w", err)
+	}
+	if err := files.WriteStringToFile(border0ConfigFile, string(yamlBytes)); err != nil {
+		return fmt.Errorf("failed to write connector token to file %s: %w", border0ConfigFile, err)
 	}
 	return nil
 }
```
Comment (adrianosela) [inline internal/connector_v2/invite/invite.go:108]: I feel like there's already a function to write the config file?

**Review by adrianosela — APPROVED** [review]
Apprved with commnt

File: internal/connector_v2/invite/invite.go (hunk 1 above)
Comment (waltzofpearls) [reply internal/connector_v2/invite/invite.go:106]: I searched the codebase, and didn't find a function to encode the config to yaml and then write to a file.

So I moved the code into connector v2 config package, and created a new function `WriteToFile`, so we can reuse it in the future.
